src/ripple/protocol/impl/TxFormats.cpp

```diff
@@ -25,6 +25,24 @@ namespace ripple {
 TxFormats::TxFormats()
 {
     // Fields shared by all txFormats:
+    static const std::initializer_list<SOElement> pseudoCommonFields{
+        {sfTransactionType, soeREQUIRED},
+        {sfFlags, soeOPTIONAL},
+        {sfSourceTag, soeOPTIONAL},
+        {sfAccount, soeREQUIRED},
+        {sfSequence, soeREQUIRED},
+        {sfPreviousTxnID, soeOPTIONAL},  // emulate027
+        {sfLastLedgerSequence, soeOPTIONAL},
+        {sfAccountTxnID, soeOPTIONAL},
+        {sfFee, soeREQUIRED},
+        {sfOperationLimit, soeOPTIONAL},
+        {sfMemos, soeOPTIONAL},
+        {sfSigningPubKey, soeREQUIRED},
+        {sfTxnSignature, soeOPTIONAL},
+        {sfSigners, soeOPTIONAL},  // submit_multisigned
+        {sfNetworkID, soeOPTIONAL},
```

Comment:
Since the only difference between the fields in `pseudo` and `common` is the ticket field, we could consider a macro to factor out the commonality. If you do this make sure you `undef` the macro after it's used.

There are other ways to do this too and some people are allergic to macros (which I understand), but I think it would be OK here. (The other way I had in mind is an immediately invoked lambda which initialized two collections and change `add` to take the collection, but the only reason to do that is to avoid the macro.)